File: src/network/packets/playerRemoved.js
import Packet from '../packet.js';
import Writer from '../utils/writer.js';
import Player from "../../ui/objects/player.js";
import {convertIntColorToHex} from "../../ui/utils.js";
import Point from "../../ui/objects/point";


class PlayerRemovedPacket extends Packet {

    constructor() {
        super();
        this.userId = null;
        this.packetId = 1010;
        this.player = null;

    }


    // Handel Server Response
    parsePacket() {
        this.userId = this.reader.readInt4();
    }

    finalize() {
        // Handle Server Request
        // Send Empty Packet As Ask For Ready
        const writer = new Writer(this.packetId);
        return writer.finalize();
    }


    handleReceivedPacket(client) {
        const player = window.gameEngine.gameObjects.players[this.userId];
        if(player)
            window.gameEngine.gameObjects.removePlayer(player);

    }
}

export default PlayerRemovedPacket;

File: gameserver/network/packets/player_removed.py
from gameserver.network.packet import Packet
from gameserver.network.utils.writer import Writer


class PlayerRemovedPacket(Packet):

    PACKET_ID = 1010

    def __init__(self, player):
        super().__init__()
        self.packet_id = self.PACKET_ID
        self.user_id = player.player_id

    def parse_packet(self):
       pass

    def handle_packet(self, client):
        pass

    def finalize(self):
        writer = Writer(self.packet_id)
        writer.write_int_in_bytes(self.user_id, 4)
        return writer.finalize()
